flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStorage.java

```diff
@@ -134,6 +149,37 @@ public CheckpointStorageLocation initializeLocationForCheckpoint(long checkpoint
 				writeBufferSize);
 	}
 
+	@Override
+	public void shutDown(JobStatus jobStatus) throws IOException {
+		if (checkpointProperties != null) {
+			boolean discardCheckpointFolders = false;
+			switch (jobStatus) {
+				case FAILED:
+					discardCheckpointFolders = checkpointProperties.discardOnJobFailed();
+					break;
+				case CANCELED:
+					discardCheckpointFolders = checkpointProperties.discardOnJobCancelled();
+					break;
+				case FINISHED:
+					discardCheckpointFolders = checkpointProperties.discardOnJobFinished();
+					break;
+				case SUSPENDED:
+					discardCheckpointFolders = checkpointProperties.discardOnJobSuspended();
+					break;
+			}
+			if (discardCheckpointFolders) {
+				fileSystem.delete(taskOwnedStateDirectory, true);
+				fileSystem.delete(sharedStateDirectory, true);
+				fileSystem.delete(checkpointsDirectory, true);
+			}
```

Comment:
```suggestion
		if (jobStatus == JobStatus.FINISHED && checkpointProperties.discardOnJobFinished() ||
			jobStatus == JobStatus.CANCELED && checkpointProperties.discardOnJobCancelled() ||
			jobStatus == JobStatus.FAILED && checkpointProperties.discardOnJobFailed() ||
			jobStatus == JobStatus.SUSPENDED && checkpointProperties.discardOnJobSuspended()) {

			doDiscard();
		}
```
Simplify the logic referring to `CompletedCheckpoint#discardOnShutdown`